### PR TITLE
Add feature flags docs

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

This came up repeatedly today in https://github.com/forem/forem/pull/12627/files#r573488069 and https://github.com/forem/forem/pull/12641#discussion_r573790187 so I went ahead and added a quick explanation about feature flags to our docs and how to use them

## QA Instructions, Screenshots, Recordings

1. cd to `docs/`
2. run `gitdocs serve`
3. view the new section under "Technical overview"

## Added tests?

- [ ] Yes
- [x] No, and this is why: documentation update
- [ ] I need help with writing tests

## Added to documentation?

- [x] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] README
- [ ] No documentation needed

### CODE REVIEW

File: app/views/layouts/_footer.html.erb [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <hr class="crayons-footer__divider">
     <p class="fs-s crayons-footer__description"><a href="/" aria-label="<%= community_name %> Home" class="crayons-link"><%= community_name %></a> – <%= SiteConfig.community_description %></p>
     <div class="m:-mb-4 crayons-footer__description">
-      <%# The following copy is necessary for compatability with the Forem AGPL licence which requires instances to link back to the source. %>
+      <%# The following copy is necessary for compatibility with the Forem AGPL licence which requires instances to link back to the source. %>
```
Comment (rhymes) [inline app/views/layouts/_footer.html.erb:22]: unrelated, but I noticed the typo

File: docs/technical-overview/compatibility.md [hunk 2]
```diff
@@ -4,9 +4,9 @@ To go along with effective administration and putting user security and
 well-being first, for a Forem to be most useful, it maximizes its technical
 compatibility.
 
-Compatability means:
+Compatibility means:
```
Comment (rhymes) [inline docs/technical-overview/compatibility.md:7]: unrelated, but I noticed the typo

**Review by joshpuetz — APPROVED** [review]
giphy

File: docs/technical-overview/feature_flags.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Feature flags
+
+We sometimes employ feature flags to develop features that aren't fully formed
+at the moment of deployment. They can also be employed to roll out features in
+productions incrementally.
```
Comment (jacobherrington) [inline docs/technical-overview/feature_flags.md:5]: {non-blocking) I'm not certain, but I think this is a typo?

```suggestion
production incrementally.
```

File: docs/technical-overview/feature_flags.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+## Enabling/disabling a feature flag globally
+
+As mentioned, feature flags can be used to test a work in progress feature or to
+test a new approach directly in production.
+
+To enable such feature, login to the Forem, go to `/admin/feature_flags` (it
+requires the `tech_admin` role), click on the flag you want to enable and press
+"Fully Enable".
```
Comment (jacobherrington) [inline docs/technical-overview/feature_flags.md:64]: (non-blocking) This seems a little hard to follow. I think presenting it as a list is easier for me to read. What do you think?

```suggestion
To enable such a feature:
- Login to the Forem
- Go to `/admin/feature_flags` (it requires the `tech_admin` role)
- Click on the flag you want to enable and press "Fully Enable".
``

**Review by jacobherrington — APPROVED** [review]
Thanks for documenting this!! I had a couple of suggestions, but I think it is good to merge even if you want to ignore my comments :smile:

**Review by citizen428 — APPROVED** [review]
👍

File: docs/technical-overview/feature_flags.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+We sometimes employ feature flags to develop features that aren't fully formed
+at the moment of deployment. They can also be employed to roll out features in
+productions incrementally.
+
+Feature flags are meant to be temporary and part of a rollout plan resulting in
+their removal.
+
+## Creating a new feature flag
+
+To create a new feature flag, we use data update scripts so that they can be
```
Comment (Ridhwana) [inline docs/technical-overview/feature_flags.md:12]: **(nitpick, non-blocking):** Should we maybe also maybe link to the data update scripts documentation here? you do explain how it works below so not critical but a nice to have I guess.

File: docs/technical-overview/feature_flags.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+made available over the entire Forem fleet, for example:
+
+```shell
+bundle exec rails g data_update AddAwesomeAlgorithmFeatureFlag
+```
+
+this will generate a script and its spec. Once done, you can add the new feature
```
Comment (Ridhwana) [inline docs/technical-overview/feature_flags.md:19]:
```suggestion
This will generate a script and its spec. Once done, you can add the new feature
```

File: docs/technical-overview/feature_flags.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    end
+  end
+end
+```
+
+and then start using it right away.
+
+## Guarding a feature behind a flag
+
+Once the feature flag is added, you can start using it to hide feature behind a
```
Comment (Ridhwana) [inline docs/technical-overview/feature_flags.md:36]:
```suggestion
Once the feature flag is added, you can start using it to hide the feature behind a
```

File: docs/technical-overview/feature_flags.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  <%= render "good_old_button" %>
+<% end %>
+```
+
+## Enabling/disabling a feature flag globally
+
+As mentioned, feature flags can be used to test a work in progress feature or to
+test a new approach directly in production.
+
+To enable such feature, login to the Forem, go to `/admin/feature_flags` (it
```
Comment (Ridhwana) [inline docs/technical-overview/feature_flags.md:62]:
```suggestion
To enable such feature, log in to the Forem, go to `/admin/feature_flags` (it
```
**(nitpick):** want probably want to use the verb here not the noun

File: docs/technical-overview/feature_flags.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+requires the `tech_admin` role), click on the flag you want to enable and press
```
Comment (Ridhwana) [inline docs/technical-overview/feature_flags.md:63]: **(suggestion, non-blocking):** Maybe a link here to the docs on how to add a role?

File: docs/technical-overview/feature_flags.md (hunk 4 above)
Comment (Ridhwana) [reply docs/technical-overview/feature_flags.md:64]: I like the list view, can we change `login` to `log in` and a suggestion is to possibly add the link to the docs on how to add a role.

**Review by Ridhwana — COMMENTED** [review]
Thank you @rhymes. This is very thorough! I just left a few comments and some suggestions 😊

File: docs/technical-overview/feature_flags.md (hunk 4 above)
Comment (rhymes) [reply docs/technical-overview/feature_flags.md]: @Ridhwana will do manually!

File: docs/technical-overview/feature_flags.md (hunk 5 above)
Comment (rhymes) [reply docs/technical-overview/feature_flags.md:12]: good idea!

File: docs/technical-overview/feature_flags.md (hunk 8 above)
Comment (rhymes) [reply docs/technical-overview/feature_flags.md]: done!

File: docs/technical-overview/feature_flags.md (hunk 9 above)
Comment (rhymes) [reply docs/technical-overview/feature_flags.md]: +1

**Review by Ridhwana — APPROVED** [review]
